### PR TITLE
feat(luna): added rate lesson dialog

### PR DESCRIPTION
**Pull request by mostafakamar2308**
I chose to make the initial value of the rating 5 as this would influence students to give perfect scores for tutors.
We can control the initial value by usign `initialRating` prop

image
image
image

### CODE REVIEW

**Comment by neuodev** [comment]
🚀 [Nova](https://nova-ohcc9s576-litespace-core.vercel.app) 
🚀 [Luna](https://luna-9fwtxf64k-litespace-core.vercel.app) 
🚀 [Nebula](https://nebula-3geqw7g5z-litespace-core.vercel.app) 
🚀 [Dashboard](https://dashboard-4h6b67jc0-litespace-core.vercel.app)

**Comment by neuodev** [comment]
🚀 [Nova](https://nova-9mrnw2any-litespace-core.vercel.app) 
🚀 [Luna](https://luna-bvdd6yvpe-litespace-core.vercel.app) 
🚀 [Nebula](https://nebula-gn4trpkfk-litespace-core.vercel.app) 
🚀 [Dashboard](https://dashboard-aixql7yxi-litespace-core.vercel.app)

File: packages/luna/src/components/Lessons/RateLesson/RateLesson.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+import { Textarea } from "@/components/Textarea";
+
+export const RateLesson: React.FC<{
+  close: Void;
+  tutorName: string | null;
+  studentId: number;
+  rateLoading: boolean;
+  initialRating?: number;
+  /**
+   * whether you are rating session or the platform itself
```
Comment (neuodev) [inline packages/luna/src/components/Lessons/RateLesson/RateLesson.tsx:18]:
```suggestion
   * Whether you are rating a session or the platform itself.
```

File: packages/luna/src/components/Lessons/RateLesson/RateLesson.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  type: "session" | "platform";
+  onRate: ({
+    id,
+    value,
+    feedback,
+  }: {
+    /**
+     * ratee id (student)
+     */
+    id: number;
```
Comment (neuodev) [inline packages/luna/src/components/Lessons/RateLesson/RateLesson.tsx:29]: No need to pass the ratee id to the `onRate` handler. The parent component is already aware of it. This is not the case for feedback and value. 

```suggestion
```

File: packages/luna/src/components/Lessons/RateLesson/RateLesson.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  studentId,
+  rateLoading,
+  type = "session",
+  initialRating,
+}) => {
+  const intl = useFormatMessage();
+  const [rating, setRating] = useState<number>(initialRating || 5);
+  const [feedback, setFeedback] = useState<string>("");
+
+  const onFeedbackchange = useCallback(
```
Comment (neuodev) [inline packages/luna/src/components/Lessons/RateLesson/RateLesson.tsx:46]:
```suggestion
  const onFeedbackChange = useCallback(
```

File: packages/luna/src/components/Lessons/RateLesson/RateLesson.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+      setFeedback(event.target.value);
+    },
+    []
+  );
+
+  const sendRating = useCallback(() => {
```
Comment (neuodev) [inline packages/luna/src/components/Lessons/RateLesson/RateLesson.tsx:53]: `send` will make the imperssion that you are sending it somewhere. This is not the case in your code. I think `setRating` is a better name.

File: packages/luna/src/components/Lessons/RateLesson/RateLesson.tsx [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+        </Typography>
+        <Typography
+          element="caption"
+          weight="regular"
+          className="tw-text-natural-950 tw-text-center"
+        >
+          {type === "session"
+            ? intl("call.rating.question.description")
+            : intl("platform.rating.question.description")}
+        </Typography>
```
Comment (neuodev) [inline packages/luna/src/components/Lessons/RateLesson/RateLesson.tsx:95]: Empty line please.

File: packages/luna/src/components/RatingStars/RatingStars.tsx [hunk 6]
```diff
@@ -71,7 +72,9 @@ export const RatingStars: React.FC<StarProps> = ({
                   : "[&>*]:tw-fill-natural-300"
               )}
               onClick={() => {
-                if (!setRating) return;
+                if (!setRating || readonly) return;
+                console.log("setting rating", idx + 1);
```
Comment (neuodev) [inline packages/luna/src/components/RatingStars/RatingStars.tsx:76]: Left over.

File: packages/luna/src/components/Textarea/Textarea.tsx [hunk 7]
```diff
@@ -7,47 +7,34 @@ export interface TextareaProps
...
+          //placeholder
+          "placeholder:tw-text-natural-400",
+          // error
+          "data-[error=true]:tw-shadow-input-error data-[error=true]:tw-border-destructive-600 data-[error=true]:hover:tw-bg-natural-50 data-[error=true]:hover:tw-border-destructive-600",
+          // disabled
+          "disabled:tw-opacity-50",
+          className
+        )}
+        {...props}
+      />
```
Comment (neuodev) [inline packages/luna/src/components/Textarea/Textarea.tsx:37]: We need to abaility to show error under that textarea. It should be the same as the input filed.

File: packages/luna/src/components/Lessons/RateLesson/RateLesson.tsx [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+import Rate from "@litespace/assets/Rate";
+import { RatingStars } from "@/components/RatingStars";
+import { Button } from "@/components/Button";
+import { Textarea } from "@/components/Textarea";
+
+export const RateLesson: React.FC<{
+  close: Void;
+  tutorName: string | null;
+  studentId: number;
+  rateLoading: boolean;
```
Comment (neuodev) [inline packages/luna/src/components/Lessons/RateLesson/RateLesson.tsx:15]: Nice!

File: packages/luna/src/components/Lessons/RateLesson/RateLesson.tsx [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+import React, { useCallback, useState } from "react";
+import Rate from "@litespace/assets/Rate";
+import { RatingStars } from "@/components/RatingStars";
+import { Button } from "@/components/Button";
+import { Textarea } from "@/components/Textarea";
+
+export const RateLesson: React.FC<{
+  close: Void;
+  tutorName: string | null;
+  studentId: number;
```
Comment (neuodev) [inline packages/luna/src/components/Lessons/RateLesson/RateLesson.tsx:14]: You don't need the student id.

**Review by neuodev — CHANGES_REQUESTED** [review]

**Comment by neuodev** [comment]
🚀 [Nova](https://nova-lmg0p5rng-litespace-core.vercel.app) 
🚀 [Luna](https://luna-ld9hn0mto-litespace-core.vercel.app) 
🚀 [Nebula](https://nebula-4x8v3kzge-litespace-core.vercel.app) 
🚀 [Dashboard](https://dashboard-dsexnizs8-litespace-core.vercel.app)

**Review by neuodev — CHANGES_REQUESTED** [review]

File: packages/luna/src/components/Lessons/RateLesson/RateLesson.tsx [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+          readonly={rateLoading}
+        />
+
+        <Textarea
+          placeholder={intl("call.rating.text.placeholder")}
+          className="tw-min-h-[138px] !tw-w-[696px]"
+          disabled={rateLoading}
+          value={feedback}
+          onChange={onFeedbackChange}
+        />
```
[the remark this answers is not in the document]
Comment (mostafakamar2308) [reply packages/luna/src/components/Lessons/RateLesson/RateLesson.tsx:103]: I think it's 24px, isn't that right?

**Comment by neuodev** [comment]
🚀 [Nova](https://nova-599tazeam-litespace-core.vercel.app) 
🚀 [Luna](https://luna-bhwf8a1ck-litespace-core.vercel.app) 
🚀 [Nebula](https://nebula-cl99sc18y-litespace-core.vercel.app) 
🚀 [Dashboard](https://dashboard-laf44x0wj-litespace-core.vercel.app)

**Comment by neuodev** [comment]
🚀 [Nova](https://nova-68e84yasg-litespace-core.vercel.app) 
🚀 [Luna](https://luna-n1vczwdgf-litespace-core.vercel.app) 
🚀 [Nebula](https://nebula-i7edthaj0-litespace-core.vercel.app) 
🚀 [Dashboard](https://dashboard-i4k1lh6w6-litespace-core.vercel.app)

File: packages/luna/src/components/Textarea/Textarea.tsx [hunk 11]
```diff
@@ -1,52 +1,68 @@
...
+        </AnimatePresence>
+
+        <AnimatePresence mode="wait" initial={false}>
+          {error ? (
+            <Helper>
+              <Typography
+                element="tiny-text"
+                className="tw-text-destructive-500"
+              >
+                {error}
```
Comment (neuodev) [inline packages/luna/src/components/Textarea/Textarea.tsx:61]: If you checked the input component clearfly, you will notice that we render the helper text here and use the the error flag to give it an error color. Please check how this is implemented.

File: packages/luna/src/components/Lessons/RateLesson/RateLesson.tsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (neuodev) [reply packages/luna/src/components/Lessons/RateLesson/RateLesson.tsx:103]: Yup, it is fixed! Thank you! 

جزاك الله خيرا يامصطفي.

**Review by neuodev — CHANGES_REQUESTED** [review]

File: packages/luna/src/components/Lessons/RateLesson/RateLesson.tsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (mostafakamar2308) [reply packages/luna/src/components/Lessons/RateLesson/RateLesson.tsx:103]: امين يارب واياكم ♥♥

**Comment by neuodev** [comment]
🚀 [Nova](https://nova-jurq32kng-litespace-core.vercel.app) 
🚀 [Luna](https://luna-cqcl7mal7-litespace-core.vercel.app) 
🚀 [Nebula](https://nebula-8m0qljhc3-litespace-core.vercel.app) 
🚀 [Dashboard](https://dashboard-avy42vcqp-litespace-core.vercel.app)

**Review by neuodev — APPROVED** [review]

**Pull request merged**